d7da184068ebf4b217db1b39be2ff973fa4abc7d
src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java
--- 
+++ 
@@ -196,7 +196,7 @@
             + " for replication messages");
 
     // Inform global state that the replication server is started, and it is safe to start indices
-    globalState.replicationStarted();
+    globalState.replicationStarted(replicationServer.getPort());
 
     LuceneServerMonitoringServerInterceptor monitoringInterceptor =
         LuceneServerMonitoringServerInterceptor.create(
@@ -740,7 +740,8 @@
                   backupFromIncArchiver,
                   restoreFromIncArchiver,
                   true,
-                  null);
+                  null,
+                  globalState.getConfiguration().getDiscoveryFileUpdateIntervalMs());
 
           indexState =
               globalState.getIndex(